File: app/src/main/java/com/example/h_mal/shift_tracker/SplashScreen.java
package com.example.h_mal.shift_tracker;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.support.v4.app.ActivityOptionsCompat;
import android.widget.RelativeLayout;

/**
 * Created by h_mal on 27/06/2017.
 */

public class SplashScreen extends Activity {

    // Splash screen timer
    private static int SPLASH_TIME_OUT = 2000;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        final Bundle bundle = ActivityOptionsCompat.makeCustomAnimation(this, R.anim.hyperspace_jump, android.R.anim.fade_out).toBundle();

        final RelativeLayout relativeLayout = (RelativeLayout) findViewById(R.id.splash_layout);

        final Intent i = new Intent(SplashScreen.this,MainActivity.class);
        i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_CLEAR_TASK);

        new Handler().postDelayed(new Runnable() {

            @Override
            public void run() {
                // This method will be executed once the timer is over
                // Start your app main activity
//                startActivity(i,bundle);
                startActivity(i);
                overridePendingTransition(android.R.anim.fade_in,R.anim.hyperspace_jump);
//                finish();
            }
        }, SPLASH_TIME_OUT);
    }


}

File: app/src/main/java/com/appttude/h_mal/farmr/ui/SplashScreen.kt
package com.appttude.h_mal.farmr.ui

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import com.appttude.h_mal.farmr.R

/**
 * Created by h_mal on 27/06/2017.
 */
@SuppressLint("CustomSplashScreen")
class SplashScreen : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        val i = Intent(this@SplashScreen, MainActivity::class.java)
        i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_CLEAR_TASK)
        Handler(Looper.getMainLooper()).postDelayed({
            startActivity(i)
            overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
            this.finish()
        }, SPLASH_TIME_OUT)
    }

    companion object {
        // Splash screen timer
        const val SPLASH_TIME_OUT: Long = 2000
    }
}